### PR TITLE
Modify conversion error, new increase format match

### LINKED ISSUE #2355: [Bug] [seatunnel-connectors] kafka source  ClassCastException: com.fasterxml.jackson.databind.node.ObjectNode cannot be cast to com.fasterxml.jackson.databind.node.ArrayNode
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

Kafka source format.type, multiple schema not matched, resulting in error: com.fasterxml.jackson.databind.node.ObjectNode cannot be cast to com.fasterxml.jackson.databind.node.ArrayNode

### SeaTunnel Version

2.1.2

### SeaTunnel Config

```conf
config/example02.conf

env {
    execution.parallelism = 1
}
source {
     KafkaTableStream {
         consumer.bootstrap.servers = "hadoop02:9092"
         consumer.group.id = "seatunnel5"
         topics = test_video
         result_table_name = test
         format.type = json
         schema = "{\"session_id\":\"string\",\"video_count\":\"int\",\"duration_time\":\"long\",\"user_id\":\"string\",\"user_age\":\"int\",\"city\":\"string\",\"session_start_time\":\"datetime\",\"session_end_time\":\"datetime\"}"
         format.ignore-parse-errors = "true"
    }
}
transform{
    sql {
        sql = "select user_id,city,user_age as age,video_count as video_sum,duration_time as max_duration_time,duration_time as min_duration_time,session_end_time as last_session_date from test"
        result_table_name = test2
    }
}
sink{
    DorisSink {
         source_table_name = test2
         fenodes = "hadoop02:8030"
         database = test_db
         table = example_user_video
         user = root
         password = anonymous123
         batch_size = 50
         doris.column_separator="\t"
         doris.columns="user_id,city,age,video_sum,max_duration_time,min_duration_time,last_session_date"
    }
}
```

### Running Command

```shell
bin/start-seatunnel-flink.sh --config config/example02.conf
```

### Error Exception

```log
Caused by: java.lang.RuntimeException: Json deserialization exception.
        at org.apache.seatunnel.common.utils.JsonUtils.parseArray(JsonUtils.java:255)
        at org.apache.seatunnel.flink.kafka.source.KafkaTableStream.prepare(KafkaTableStream.java:114)
        at org.apache.seatunnel.flink.kafka.source.KafkaTableStream.prepare(KafkaTableStream.java:51)
        at org.apache.seatunnel.core.base.command.BaseTaskExecuteCommand.lambda$prepare$0(BaseTaskExecuteCommand.java:66)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at org.apache.seatunnel.core.base.command.BaseTaskExecuteCommand.prepare(BaseTaskExecuteCommand.java:66)
        at org.apache.seatunnel.core.flink.command.FlinkApiTaskExecuteCommand.execute(FlinkApiTaskExecuteCommand.java:81)
        ... 18 more
Caused by: java.lang.ClassCastException: com.fasterxml.jackson.databind.node.ObjectNode cannot be cast to com.fasterxml.jackson.databind.node.ArrayNode
        at org.apache.seatunnel.common.utils.JsonUtils.parseArray(JsonUtils.java:253)
```

### Flink or Spark Version

flink v1.12.7

### Java or Scala Version

java 1.8
scala 2.12.15

### Screenshots

error

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hanwei996**
## Purpose of this pull request
https://github.com/apache/incubator-seatunnel/issues/2355
https://github.com/apache/incubator-seatunnel/issues/2313#issuecomment-1200536758   :  The problem has not been addressed at its root
kafka source ClassCastException: com.fasterxml.jackson.databind.node.ObjectNode cannot be cast to com.fasterxml.jackson.databind.node.ArrayNode

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-kafka/src/main/java/org/apache/seatunnel/flink/kafka/source/KafkaTableStream.java [hunk 1]
```diff
@@ -110,8 +110,19 @@ public void prepare(FlinkEnvironment env) {
...
-        schemaInfo = JsonUtils.parseArray(schemaContent);
+        String sourceFormat = config.getString(SOURCE_FORMAT).trim().toLowerCase();
+        format = FormatType.from(sourceFormat);
+        switch (sourceFormat){
+            case "csv":
+                schemaInfo = JsonUtils.parseArray(schemaContent);
+                break;
+            case "json":
+            case "avro":
+                schemaInfo = JsonUtils.parseObject(schemaContent);
```
Comment (Hisoka-X) [inline seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-kafka/src/main/java/org/apache/seatunnel/flink/kafka/source/KafkaTableStream.java:123]: The method only support schema are object when use `json`. Can you support array in `json` schema both?

Comment (hanwei996) [reply seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-kafka/src/main/java/org/apache/seatunnel/flink/kafka/source/KafkaTableStream.java:123]: Yes, I can.

Comment (hanwei996) [reply seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-kafka/src/main/java/org/apache/seatunnel/flink/kafka/source/KafkaTableStream.java:123]: @Hisoka-X  The new PR has been submitted

Comment (Hisoka-X) [reply seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-kafka/src/main/java/org/apache/seatunnel/flink/kafka/source/KafkaTableStream.java:123]: > @Hisoka-X The new PR has been submitted

Seem like you should fix checkstyle

Comment (hanwei996) [reply seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-kafka/src/main/java/org/apache/seatunnel/flink/kafka/source/KafkaTableStream.java:123]: @Hisoka-X  Try again

File: seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/util/SchemaUtil.java [hunk 2]
```diff
@@ -117,8 +120,27 @@ public static FormatDescriptor setFormat(FormatType format, Config config) throw
         return formatDescriptor;
     }
 
-    private static void getJsonSchema(Schema schema, ObjectNode json) {
-        Iterator<Map.Entry<String, JsonNode>> nodeIterator = json.fields();
+    private static void getJsonSchema(Schema schema, String json) {
```
Comment (Hisoka-X) [inline seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/util/SchemaUtil.java:123]: Sorry, but I don't think this method can implement schema parsing in list form. Although the fields in the list are processed cyclically, the returned schema does not reflect the attributes of the list. Maybe a field can be added for this situation: `list`, which points to the list at the outermost level of the schema

Comment (hanwei996) [reply seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/util/SchemaUtil.java:123]: Yes, there is a problem. Let me fix it

Comment (hanwei996) [reply seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/util/SchemaUtil.java:123]: The Flink schema is not familiar enough to know if this is the right thing to do
image

Comment (hanwei996) [reply seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/util/SchemaUtil.java:123]: @Hisoka-X

Comment (Hisoka-X) [reply seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/util/SchemaUtil.java:123]: Sorry for late reply, you are right, same as flink.

**Comment by Hisoka-X** [comment]
Hi, Any update or need help？

**Comment by EricJoy2048** [comment]
This pr is closed because it has not responded for too long. If necessary, please reopen it

**Pull request closed without merging**